src/components/media-loader.js

```diff
@@ -48,14 +62,26 @@ AFRAME.registerComponent("media-loader", {
   },
 
   showLoader() {
-    const loadingObj = new THREE.Mesh(new THREE.BoxGeometry(), new THREE.MeshBasicMaterial());
+    const useFancyLoader = !!loadingObject;
+    const loadingObj = useFancyLoader
+      ? loadingObject.scene.clone()
+      : new THREE.Mesh(new THREE.BoxGeometry(), new THREE.MeshBasicMaterial());
+    if (useFancyLoader) {
+      this.loaderMixer = new THREE.AnimationMixer(loadingObj);
+      this.loadingClip = this.loaderMixer.clipAction(loadingObject.animations[0]);
+      this.loadingClip.play();
+    }
     this.el.setObject3D("mesh", loadingObj);
     this.setShapeAndScale(true);
     delete this.showLoaderTimeout;
   },
 
   clearLoadingTimeout() {
     clearTimeout(this.showLoaderTimeout);
+    if (this.loaderMixer) {
+      this.loadingClip.stop();
+      this.loaderMixer = null;
```

Comment:
I would use `delete this.loadingMixer` instead. Doesn't actually make a difference in your code but I think the intend reads more cleanly.

Reply:
👍